Modules/SubModule_Structural/xASL_wrp_GetVolumetrics.m

```diff
@@ -94,21 +94,29 @@ function xASL_wrp_GetVolumetrics(x)
     xASL_adm_DeleteFileList(x.D.TissueVolumeDir, '^LST_tlv_.*\.(csv|tsv)$', [], [0 Inf]);
     xASL_adm_DeleteFileList(x.D.TissueVolumeDir, ['WMH_LST_(LGA|LPA)_' x.P.SubjectID '.(csv|tsv)'], [], [0 Inf]);
     
-    % run the LST lesion volume calculator
+    % Run the LST lesion volume calculator
+	% Change directory to the SUBJECT DIRECTORY as the count function outputs to the current directory
     CurrDir = pwd;
-    cd(x.D.TissueVolumeDir);
+    cd(x.dir.SUBJECTDIR);
     [~, FileName] = ps_LST_tlv(x.P.Path_WMH_SEGM, ~bVerbose, BinThresh, MinimalLesionVolume);
     cd(CurrDir);
     % get current filename:
-    FileName = fullfile(x.D.TissueVolumeDir, FileName);
+    FileName = fullfile(x.dir.SUBJECTDIR, FileName);
     % define new filename:
-    OutputPath = fullfile(x.D.TissueVolumeDir, ['WMH_LST_' x.WMHsegmAlg '_' x.P.SubjectID '.csv']);
-    % Then move our file
+    OutputPathCSV = fullfile(x.dir.SUBJECTDIR, ['WMH_LST_' x.WMHsegmAlg '_' x.P.SubjectID '.csv']);
```

Comment:
From here below you need to revert, it is too long and wrong; this doesn't concern temporary files, and the output should remain in the population folder.

We can eventually move all structural output to subject folders, but that would be a large future issue.

Reply:
Actually. It has to stay like this. The temporary files in population folder were completely messed up during parallelization. As this is structural module outputting to population folder. If you still want to output temp files there, then you need to change the entire structure and debug. I don't see issue outputting temp files to subject folder.

Reply:
No, I agreed with the first part where you change the temporary file to the subject folder (assuming LST outputs the same name for each subject, which gets messed up during parallelization).

It was just the last part where you added unnecessary code.

Reply:
OK. Now I understand! But there's still an issue.
1. The former version saved the temporary to POPULATION dir and then renamed it - this was causing issues in parallelization.
2. So I made a new version that saved the temporary to SUBJECTDIR and then moved the final to POPULATION. But I agree that it was way to complicated.
3. But you edit saves everything to SUBJECTDIR and never moves it to the population.
4. So I made a last fix that saves temporary to SUBJECTDIR and then moves it to POPULATION. But please check if this is clean enough.